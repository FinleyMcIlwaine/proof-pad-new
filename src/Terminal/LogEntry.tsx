/**
 * @license
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";

export enum Kind {
  INPUT,
  OUTPUT,
}

export class LogEntry extends React.Component<
  { kind: Kind; value: string },
  { expanded: boolean }
> {
  private elem: HTMLDivElement | null = null;
  state = { expanded: true };

  componentDidMount() {
    if (
      this.elem &&
      this.elem.clientHeight > 150 &&
      this.props.kind === Kind.OUTPUT
    ) {
      this.setState({ expanded: false });
    }
  }

  render() {
    return (
      <div
        ref={d => {
          this.elem = d;
        }}
      >
        <div
          style={{
            overflow: "hidden",
            maxHeight: this.state.expanded ? "" : 100,
          }}
        >
          <span className="display">
            {this.props.kind === Kind.INPUT ? ">" : "<"}&nbsp;
          </span>
          {this.props.value}
        </div>
        <div>
          {this.state.expanded ? (
            ""
          ) : (
            <button
              className="show-more"
              onClick={() => {
                this.setState({ expanded: true });
              }}
            >
              Show more...
            </button>
          )}
        </div>
      </div>
    );
  }
}
