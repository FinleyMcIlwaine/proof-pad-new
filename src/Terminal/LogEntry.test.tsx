/**
 * @license
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { LogEntry, Kind } from "./LogEntry";
import { create } from "react-test-renderer";

test("Renders inputs", () => {
  const c = create(<LogEntry kind={Kind.INPUT} value="hi" />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("Renders small outputs", () => {
  const c = create(<LogEntry kind={Kind.OUTPUT} value="bye" />);
  expect(c.toJSON()).toMatchSnapshot();
});

// There's no dom, so this test can't actually check the div's height.
test("Renders big outputs", () => {
  const c = create(<LogEntry kind={Kind.OUTPUT} value="bye" />);
  c.root.instance!.setState({ expanded: false });
  expect(c.toJSON()).toMatchSnapshot();
});
