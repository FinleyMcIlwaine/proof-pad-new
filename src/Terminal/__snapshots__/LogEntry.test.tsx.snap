// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders big outputs 1`] = `
<div>
  <div
    style={
      Object {
        "maxHeight": 100,
        "overflow": "hidden",
      }
    }
  >
    <span
      className="display"
    >
      &lt;
       
    </span>
    bye
  </div>
  <div>
    <button
      className="show-more"
      onClick={[Function]}
    >
      Show more...
    </button>
  </div>
</div>
`;

exports[`Renders inputs 1`] = `
<div>
  <div
    style={
      Object {
        "maxHeight": "",
        "overflow": "hidden",
      }
    }
  >
    <span
      className="display"
    >
      &gt;
       
    </span>
    hi
  </div>
  <div>
    
  </div>
</div>
`;

exports[`Renders small outputs 1`] = `
<div>
  <div
    style={
      Object {
        "maxHeight": "",
        "overflow": "hidden",
      }
    }
  >
    <span
      className="display"
    >
      &lt;
       
    </span>
    bye
  </div>
  <div>
    
  </div>
</div>
`;
